corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -447,6 +450,69 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
             }
         },
 
+        columnStyle: function () {
+            let self = this;
+            let mapColumn = "";
+            if (self.showMap && !self.hasNoItems) {
+                mapColumn = "[map] 300px";
+            }
+            return "display: grid;grid-template-columns: [tiles] auto " + mapColumn
+                + " ;grid-template-rows: auto";
+        },
+
+        addAddressPin: function (geocoder, addressMap, headers, model, addressIndex) {
+            let address = model.attributes.data[addressIndex];
+            geocoder.query(address, function (err, data) {
+                if (err === null) {
+                    let popupText = "";
+                    headers.forEach((header, index) => {
+                        if (header) {
+                            popupText += "<b>" + header + ":</b> " + model.attributes.data[index] + "<br>";
+                        }
+                    });
+                    L.marker(data.latlng)
+                        .addTo(addressMap)
+                        .bindPopup(popupText);
+                }
+            });
+        },
+
+        loadMap: function () {
+            let token = initialPageData.get("mapbox_access_token");
+
+            try {
+                let lat = 30;
```

Comment:
Can you file a ticket or note somewhere to update this to center the map based on the given data? (Or does the mapped data affect how the map is centered?)

Reply:
Couldn't help myself. Learned about promises and added code to fit all the pin on the map.